src/actions.ts

```diff
@@ -0,0 +1,42 @@
+import * as core from '@actions/core'
+import * as spin from './spin'
+import * as github from './github'
+import * as cloud from './cloud'
+
+export async function setup(): Promise<void> {
+    let version = core.getInput('version');
+    if (!version || version === 'latest') {
+        version = await github.getLatestRelease()
+    }
+
+    await spin.install(version)
+
+    //todo: check compatibility with spin version
+    const pluginsList = core.getInput('plugins') !== '' ? core.getInput('plugins').split(',') : [];
+    if (pluginsList.length > 0) {
+        await spin.installPlugins(pluginsList)
+    }
+}
+
+export async function deploy(): Promise<cloud.Metadata> {
+    const manifestFile = getManifestFile()
+    const cloudBaseURL = core.getInput('cloud_base_url') || cloud.DEFAULT_CLOUD_URL
```

Comment:
`cloud_base_url` isn't listed as an input - is it a 'hidden' input, or is this line obsolete?

Reply:
i am inclined toward removing it altogether. we can always add it back in future if required.